crates/holochain_types/src/app.rs

```diff
@@ -270,7 +270,7 @@ pub enum RoleSettingsYaml {
 #[serde(tag = "type", content = "value", rename_all = "snake_case")]
 pub enum AppBundleSource {
     /// The actual serialized bytes of a bundle
```

Comment:
```suggestion
    /// The raw bytes of an app bundle
```

I'm not sure what is meant by "serialized" here -- I thought this is just the raw bytes of a .happ file right?

Reply:
An app bundle source serializes to some object format that has paths in. It encodes to a msgpacked+gzipped archive containing your DNAs (assuming you bundled them). I believe this was a mistake by the implementor and it's not really usable in its current form unless you want to unpack a hApp first and place the DNAs at appropriate paths before asking Holochain to do the install.

Happy to be wrong about this, but that's my understanding of what we're trying to fix here

Reply:
>  It encodes to a msgpacked+gzipped archive containing your DNAs (assuming you bundled them)

Right, and this is written to the `.happ` file when you run `hc app pack ...`

I just find the doc comment confusing. An app bundle *is* an encoded format. The variants are either the raw bytes of the bundle or a path to a bundle file.

Reply:
Done: https://github.com/holochain/holochain/pull/4745/commits/219f8e1a4c9d53f976f8215b14f2ec850cd4217e.